sql/src/main/codegen/includes/insert.ftl

```diff
@@ -36,6 +36,11 @@ SqlNode DruidSqlInsertEof() :
     <CLUSTERED> <BY>
     clusteredBy = ClusterItems()
   ]
+  {
+      if (clusteredBy != null && partitionedBy.lhs == null) {
+        throw new ParseException("CLUSTERED BY clause found before PARTITIONED BY, the PARITITIONED BY clause has to be specified first before the CLUSTERED BY clause.");
+      }
+  }
```

Comment:
@LakshSingla if this approach looks alright, it would be possible to move some of the other checks here as well.

Reply:
This does seem alright, my one concern is that .ftl files are not supported by default (or using free plugins) in Intellij (community edition). As we move more java code to ftl, it might be difficult to keep a check on them. Is that something that we should take into consideration?

Reply:
what is the advantage of keeping a check here instead of keeping it in the java code?

Reply:
Since this is an error/check on the parsing side, it seems more in tune with Calcite's Parser.jj where the parsing errors are thrown from the .jj file itself. From a functional standpoint, they are equivalent. (It might be beneficial to throw the error before creating a node in some cases, but I don't think they apply here)

Reply:
Partitioned by is nullable in the insert and replace classes, we can make that a non null to make it cleaner if we move the null check here.